test/fastly/client_test.rb

```diff
@@ -29,6 +29,15 @@
       assert_equal "Invalid auth credentials. Check username/password.", e.message
     end
 
+    it 'surfaces a deprecation message when a username or password is provided' do
+      stub_request(:any, /api.fastly.com/).
+        to_return(body: JSON.generate(i: "dont care"), status: 200)
+
+      client = Fastly::Client.new(user: user, password: pass)
+      
+      assert_output('', /DEPRECATION WARNING:/) { $stderr.puts 'DEPRECATION WARNING:'}
```

Comment:
This test isn't really asserting anything unless you put the `Fastly::Client.new` invocation inside of the block.

```suggestion
      assert_output('', /DEPRECATION WARNING:/) { Fastly::Client.new(user: user, password: pass) }
```